src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt

```diff
@@ -0,0 +1,148 @@
+//      "ndmisReferralsPerformanceReportWriter",
+//      FileSystemResource(path),
+//      ReferralsData.headers,
+//      ReferralsData.fields
+//    )
+//  }
+//
+//  @Bean
+//  @StepScope
+//  fun ndmisComplexityWriter(@Value("#{jobExecutionContext['output.file.path']}") path: String): FlatFileItemWriter<ComplexityData> {
+//    return batchUtils.csvFileWriter(
+//      "ndmisComplexityPerformanceReportWriter",
+//      FileSystemResource(path),
+//      ComplexityData.headers,
+//      ComplexityData.fields
+//    )
+//  }
+//
+//  @Bean
+//  @StepScope
+//  fun ndmisAppointmentWriter(@Value("#{jobExecutionContext['output.file.path']}") path: String): FlatFileItemWriter<AppointmentData> {
+//    return batchUtils.csvFileWriter(
+//      "ndmisAppointmentPerformanceReportWriter",
+//      FileSystemResource(path),
+//      AppointmentData.headers,
+//      AppointmentData.fields
+//    )
+//  }
+
+  @Bean
+  fun ndmisPerformanceReportJob(
+    ndmisWriteReferralToCsvStep: Step,
+    ndmisWriteComplexityToCsvStep: Step,
+    ndmisWriteAppointmentToCsvStep: Step
+  ): Job {
+    val validator = DefaultJobParametersValidator()
+    validator.setRequiredKeys(
+      arrayOf(
+        "timestamp",
+      )
+    )
+
+    return jobBuilderFactory["ndmisPerformanceReportJob"]
+      .validator(validator)
+//      .listener(listener)
+      .start(ndmisWriteReferralToCsvStep)
+      .next(ndmisWriteComplexityToCsvStep)
+      .next(ndmisWriteAppointmentToCsvStep)
+      .build()
+  }
+
+  @Bean
+  fun ndmisWriteReferralToCsvStep(
+    ndmisReader: HibernateCursorItemReader<Referral>,
+    processor: NdmisReferralsPerformanceReportProcessor,
+//    writer: FlatFileItemWriter<ReferralsData>,
+  ): Step {
+    return stepBuilderFactory.get("ndmisWriteReferralToCsvStep")
+      .chunk<Referral, ReferralsData>(chunkSize)
+      .reader(ndmisReader)
+      .processor(processor)
+      .writer(LoggingWriter<ReferralsData>())
+      .build()
+  }
+
+  @Bean
+  fun ndmisWriteComplexityToCsvStep(
+    ndmisReader: HibernateCursorItemReader<Referral>,
+    processor: NdmisComplexityPerformanceReportProcessor,
+//    writer: FlatFileItemWriter<ComplexityData>,
+  ): Step {
+    return stepBuilderFactory.get("ndmisWriteComplexityToCsvStep")
+      .chunk<Referral, List<ComplexityData>>(chunkSize)
+      .reader(ndmisReader)
+      .processor(processor)
+      .writer(LoggingWriter<List<ComplexityData>>())
+      .build()
+  }
+
+  @Bean
+  fun ndmisWriteAppointmentToCsvStep(
+    ndmisReader: HibernateCursorItemReader<Referral>,
+    processor: NdmisAppointmentPerformanceReportProcessor,
+//    writer: FlatFileItemWriter<AppointmentData>,
+  ): Step {
+    return stepBuilderFactory.get("ndmisWriteAppointmentToCsvStep")
+      .chunk<Referral, List<AppointmentData>>(chunkSize)
+      .reader(ndmisReader)
+      .processor(processor)
+      .writer(LoggingWriter<List<AppointmentData>>())
+      .build()
+  }
+}
+
+class LoggingWriter<T> : ItemWriter<T> {
```

Comment:
temporary writer used to log results of rows. Will be replaced when we write to a csv.

Reply:
let's add this to batch utils as a permanent convenience class for debugging new jobs